kafka/src/main/java/org/apache/calcite/adapter/kafka/KafkaRowConverter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.kafka;
+
+import org.apache.calcite.rel.type.RelDataType;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+/**
+ * Interface to handle formatting between Kafka message and Calcite row.
+ *
+ * @param <K>: type for Kafka message key,
+ *           refer to {@link ConsumerConfig#KEY_DESERIALIZER_CLASS_CONFIG};
+ * @param <V>: type for Kafka message value,
+ *           refer to {@link ConsumerConfig#VALUE_DESERIALIZER_CLASS_CONFIG};
+ *
+ */
+public interface KafkaRowConverter<K, V> {
+
+  /**
+   * Generate row type for a given Kafka topic.
+   *
+   * @param topicName, Kafka topic name;
+   * @return row type
+   */
+  RelDataType rowDataType(String topicName);
```

Comment:
From configuration I see that you define one converter per topic. If so, why do you need this method (`rowDataType(String)`) ?

Reply:
I don't put row schema in constructor as `KafkaRowConverter` is defined as interface. Usually we use an external metadata system to manage Kafka message schema, calling `rowDataType(String topic)` would be more clear IMO.

Reply:
I feel like current interface is both one-to-one (between topic and row) and one-to-many (between topic and `RelDataType`). It seems more logical to have the following: 
```java
interface KafkaRowConverter {
   RelDataType rowDataType();
   Object[] toRow(ConsumerRecord<K, V> message);
}
```

Ie have separate KafkaRowConverter _per topic_. 

Another option is to provide `RelDataType` as constructor argument to `KafkaTable`.

Reply:
Traditionally we have been using single `Function1<Consumer<K,V>, Object[]>` for such transformations.

Reply:
@asereda-gs sry that I was lost in the thread, the proposed interface looks good to me. If it's not a block issue, I would like to close this PR and update it in a new task([CALCITE-3080](https://issues.apache.org/jira/browse/CALCITE-3080)), to handle user-specified columns together. In this way, there would be two provided implementations: 1) to handle user-specified columns, 2) to support external row metadata;

Reply:
OK if  [CALCITE-3080](https://issues.apache.org/jira/browse/CALCITE-3080) addresses it separately.